java/configuration/src/main/java/sleeper/configuration/properties/SleeperProperty.java

```diff
@@ -27,11 +27,17 @@ public interface SleeperProperty {
 
     String getDefaultValue();
 
+    String getDescription();
+
     default Predicate<String> validationPredicate() {
         return (s) -> true;
     }
 
     default String toEnvironmentVariable() {
         return getPropertyName().toUpperCase(Locale.ROOT).replace('.', '_');
     }
+
+    default String formatDescription() {
+        return "# " + getDescription().replace("\n", "\n# ");
+    }
```

Comment:
This format seems specific to the clients, and the context in which we display the description. Maybe this should be in the clients module?